src/main/java/com/google/sps/utility/GeoHash.java

```diff
@@ -0,0 +1,198 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Geohash encoding/decoding and associated functions   (c) Chris Veness 2014-2019 / MIT Licence
+// https://www.movable-type.co.uk/scripts/geohash.html
+// Originally in JS, transcribed to Java and added TRIM function
+//
+// Geohash: Gustavo Niemeyer’s geocoding system.
+
+package com.google.sps.utility;
+
+import com.google.sps.COMMONS;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+
+public class GeoHash {
+
+  /**
+   * Encodes latitude and longitude to geoHash given determined distance to set the
+   * precision of the hash and creates a list of geohashed neighbour cells in order
+   * to create a valid query list.
+   * @param lat {double} Latitude in degrees.
+   * @param lng {double} longitude in degrees.
+   * @param distance {int} distance to determine precision in meters
+   * @return {List<String>} List of adjacent geohashes of supplied latitude/longitude.
+   */
+  public static List<String> getHashesToQuery(double lat, double lng, int distance) {
+    String geohash = encode(lat, lng, distance);
+    List<String> hashesToQuery = neighbourCells(geohash);
+
+    // Append center geohash cell (current)
+    hashesToQuery.add(geohash);
+
+    return hashesToQuery;
+  }
+
+  /**
+   * Calls the function encode with distance of 0 that will set the precision
+   * to the maximum (9). That value will be stored on datastore.
+   * @param lat {double} latitude of the Vendor.
+   * @param lng {double} longitude of the Vendor.
+   * @return {String} encoded geoHashed.
+   */
+  public static String encodeVendor(double lat, double lng) {
+    return encode(lat, lng, 0);
+  }
+
+  /**
+   * Encodes latitude and longitude to geoHash given determined distance to set the
+   * precision of the hash.
+   * @param lat {double} Latitude in degrees.
+   * @param lng {double} longitude in degrees.
+   * @param distance {int} distance to determine precision in meters
+   * @return {string} Geohash of supplied latitude/longitude.
+   */
+  private static String encode(double lat, double lng, int distance) {
+    String geoHash = "";
+    int precision = determinePrecision(distance);
+    int index = 0; // index into BASE32 map.
+    int bit = 0; // each char holds 5 bits.
+    boolean evenBit = true;
+    double latMin = -90, latMax = 90;
+    double lngMin = -180, lngMax = 180;
+
+    while (geoHash.length() < precision) {
```

Comment:
Can you add some comments to see whats going on inside this while statement? Just to know how the encoding operations work :sweat_smile:

Reply:
+1, this code is a bit difficult to read on its own, comments would be helpful!